apis/submariner/v1alpha1/submariner_types.go

```diff
@@ -124,8 +124,54 @@ type SubmarinerList struct {
 	Items           []Submariner `json:"items"`
 }
 
+// BrokerSpec defines the desired state of Broker
+// +k8s:openapi-gen=true
+type BrokerSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	Components                  []string `json:"components,omitempty"`
+	DefaultCustomDomains        []string `json:"defaultCustomDomains,omitempty"`
+	GlobalnetCidrRange          string   `json:"globalnetCidrRange,omitempty"`
+	DefaultGlobalnetClusterSize uint     `json:"defaultGlobalnetClusterSize"`
+	GlobalnetEnabled            bool     `json:"globalnetEnabled"`
```

Comment:
Can we make GlobalNetEnabled an optional/omitempty argument?, same for DefaultGlobalnetClusterSize?, is it possible -an alternative- to provide default values and allow omission?

Is it really necessary?, could it be that "GlobalnetCidrRange == GlobalnetEnabled?" (not sure about it though).

Reply:
How about making Globalnet a component? I’ll try that in a separate PR... Also, Globalnet can be enabled without a CIDR range, can’t it?

Reply:
Hmm, that'd be rather neat I think @skitt 

Without a CIDR range I'm not sure, I guess we need something at least for auto-assignment of clusters. But if you don't want auto-assignment I guess you don't need it @sridhargaddam  ^

Reply:
Correct, the globalCIDR is required for auto-assignment to clusters while the cluster is joining the Broker. We provide flexibility to users to specify custom globalCIDR, but IMO we should continue to have a default CIDR to support clusters that do not explicitly specify globalCIDR while joining.

Reply:
The fields are now all optional.

Reply:
https://github.com/submariner-io/submariner-operator/issues/1102 tracks the componentisation of Globalnet.